src/engine/controls/enginecontrol.h

```diff
@@ -48,17 +48,17 @@ class EngineControl : public QObject {
             UserSettingsPointer pConfig);
     ~EngineControl() override;
 
-    // Called by EngineBuffer::process every latency period. See the above
-    // comments for information about guarantees that hold during this call. An
-    // EngineControl can perform any upkeep operations that are necessary during
-    // this call.
+    /// Called by EngineBuffer::process every latency period.
+    /// See the above comments for information about guarantees that hold during this call.
+    /// An EngineControl can perform any upkeep operations necessary here.
+    /// @param dRate current playback rate in audio frames per second
     virtual void process(const double dRate,
```

Comment:
```suggestion
    virtual void process(const mixxx::audio::SampleRate rate,
```
(also in the implementation)

Reply:
`mixxx::audio::SampleRate` is an `unsigned int`, but this rate can become negative, if we play reverse.